### PR TITLE
Fixing memory leak caused by blocking query

### PR DESCRIPTION
**Pull request by armon**
Blocking queries work by registering a notification channel along various write paths and watching for trigger to be hit. Once triggered, those channels are notified and cleared. However, for many blocking queries, their trigger point may not be hit for a long (potentially infinite) time, causing a memory leak. To avoid this, this PR changes behavior to pro-actively clear the notification channels and timeout timers. This avoids the leaks caused by blocking queries.

### CODE REVIEW

File: consul/state_store.go [hunk 1]
```diff
@@ -421,6 +421,13 @@ func (s *StateStore) Watch(tables MDBTables, notify chan struct{}) {
 	}
 }
 
+// StopWatch is used to unsubscribe a channel to a set of MDBTables
+func (s *StateStore) StopWatch(tables MDBTables, notify chan struct{}) {
```
Comment (ryanuber) [inline consul/state_store.go]: I think we are missing a test for this function

**Comment by ryanuber** [comment]
@armon everything looks great, just one untested method. Otherwise LGTM!

**Comment by armon** [comment]
@ryanuber Added test!

**Pull request merged**